apps/examples/testcase/le_tc/kernel/tc_libc_spawn.c

```diff
@@ -381,13 +435,23 @@ static void tc_libc_spawn_posix_spawn_file_actions_dump(void)
 	ret_chk = posix_spawn_file_actions_init(&st_fileactions);
 	TC_ASSERT_EQ("posix_spawn_file_actions_init", ret_chk, OK);
 
+	posix_spawn_file_actions_dump(&st_fileactions);
+
 	ret_chk = posix_spawn_file_actions_addopen(&st_fileactions, 1, szfilepath, O_WRONLY, 0644);
 	TC_ASSERT_EQ("posix_spawn_file_actions_addopen", ret_chk, OK);
 
 	/* posix_spawn_file_actions_dump returns (void) */
 
 	posix_spawn_file_actions_dump(&st_fileactions);
 
+	ret_chk = posix_spawn_file_actions_adddup2(&st_fileactions, 1, 2);
+	TC_ASSERT_EQ("posix_spawn_file_actions_adddup2", ret_chk, OK);
+	posix_spawn_file_actions_dump(&st_fileactions);
+
+	ret_chk = posix_spawn_file_actions_addclose(&st_fileactions, 1);
```

Comment:
if posix_spawn_file_actions_addclose is failed, who will free the mem from posix_spawn_file_actions_adddup2?
I mean, in posix_spawn_file_actions_adddup2, there is a mem allocation.

Reply:
posix_spawn_file_actions_destroy(&st_fileactions); 
will take care of that.
It does not fail once memory is allocated as it just adds it to the file actions list.
Reference:
tc_libc_spawn_posix_spawn_file_actions_addopenclose(void)
In this posix_spawn_file_actions_addclose() API also allocates memory which is further freed using destroy API.

Reply:
I think posix_spawn_file_actions_addclose can be failed.
if mem is not available, posix_spawn_file_actions_addclose will be fail.
in this case, posix_spawn_file_actions_adddup2 is not freed.
isn't it?

Reply:
Yes, correct..
We can follow 2 approaches:
Add a label in case of failure...like >>
TC_ASSERT_EQ_CLEANUP("posix_spawn_file_actions_adddup2", ret_chk, OK, goto cleanup);

cleanup:     ret_chk = posix_spawn_file_actions_destroy(&st_fileactions);
                   TC_ASSERT_EQ("posix_spawn_file_actions_destroy", ret_chk, OK);

OR
After every addclose/adddup2/addopen call destroy API
ret_chk = posix_spawn_file_actions_adddup2(&st_fileactions, 1, 2);
TC_ASSERT_EQ("posix_spawn_file_actions_adddup2", ret_chk, OK);

ret_chk = posix_spawn_file_actions_destroy(&st_fileactions);
TC_ASSERT_EQ("posix_spawn_file_actions_destroy", ret_chk, OK);

ret_chk = posix_spawn_file_actions_addclose(&st_fileactions, 1);
TC_ASSERT_EQ("posix_spawn_file_actions_addclose", ret_chk, OK);

ret_chk = posix_spawn_file_actions_destroy(&st_fileactions);
TC_ASSERT_EQ("posix_spawn_file_actions_destroy", ret_chk, OK);

Reply:
I think 1st is better.

Reply:
ok will update